src/backend/storage/smgr/smgr.c

```diff
@@ -419,34 +387,9 @@ smgrclosenode(RelFileNodeBackend rnode)
 void
 smgrcreate(SMgrRelation reln, ForkNumber forknum, bool isRedo)
 {
-<<<<<<< HEAD
-	/*
-	 * Exit quickly in WAL replay mode if we've already opened the file. If
-	 * it's open, it surely must exist.
-	 */
-	if (isRedo && reln->md_num_open_segs[forknum] > 0)
-		return;
-
-	/*
-	 * We may be using the target table space for the first time in this
-	 * database, so create a per-database subdirectory if needed.
-	 *
-	 * XXX this is a fairly ugly violation of module layering, but this seems
-	 * to be the best place to put the check.  Maybe TablespaceCreateDbspace
-	 * should be here and not in commands/tablespace.c?  But that would imply
-	 * importing a lot of stuff that smgr.c oughtn't know, either.
-	 */
-	TablespaceCreateDbspace(reln->smgr_rnode.node.spcNode,
-							reln->smgr_rnode.node.dbNode,
-							isRedo);
-
-    (*reln->storageManager).smgr_create(reln, forknum, isRedo);
-
 	if (file_create_hook)
 		(*file_create_hook)(reln->smgr_rnode);
-=======
 	smgrsw[reln->smgr_which].smgr_create(reln, forknum, isRedo);
->>>>>>> sync-pg-phase1
 }
```

Comment:
Previously, we called `file_create_hook` after calling `smgr_create`. Now we do it before. Won't that break the logic?

Reply:
moved